viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx

```diff
@@ -196,14 +231,14 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         }
     }
 
-    private async fetchTimegraphData(range: TimelineChart.TimeGraphRange, resolution: number) {
+    private async fetchTimegraphData(ids: number[], range: TimelineChart.TimeGraphRange, resolution: number) {
         const length = range.end - range.start;
```

Comment:
Talking to Soukaina again:

In this method, you could get the ids using this.state.timegraphEntry and then with the 2 lines from componentDidUdate:

 const treeNodes = listToTree(treeResponse.model.entries);
 const orderedTreeIds = getAllVisibleNodeIds(treeNodes, []);

You have the orderedTreeIds. This way, you don't need the ids in parameter, the chart layer can be created in the constructor.

The only thing remaining is to trigger a refresh of the time graph chart on collapse (state change)